client-cli/src/test/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationUrlResolverTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.registration.cli;
+
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.Service;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidResolver;
+import org.eclipse.dataspaceconnector.iam.did.web.resolution.WebDidResolver;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class RegistrationUrlResolverTest {
+
+    public static final String SERVICE_TYPE = "RegistrationUrl";
+
+    DidResolver didResolver = mock(WebDidResolver.class);
+    RegistrationUrlResolver urlResolver = new RegistrationUrlResolver(didResolver);
+
+    String did = "did:web:didserver";
+
+    @Test
+    void resolveUrl_success() {
+
+        String apiUrl = "http://registrationUrl/api";
+        DidDocument didDocument = didDocument(apiUrl);
+        when(didResolver.resolve(did)).thenReturn(Result.success(didDocument));
+
+        Optional<String> resultApiUrl = urlResolver.resolveUrl(did);
+
+        assertThat(resultApiUrl).isNotEmpty();
+        assertThat(resultApiUrl.get()).isEqualTo(apiUrl);
+
+    }
+
+    @Test
+    void resolveUrl_noRegistrationUrl() {
+
+        DidDocument didDocument = DidDocument.Builder.newInstance().build();
+        when(didResolver.resolve(did)).thenReturn(Result.success(didDocument));
+
+        Optional<String> resultApiUrl = urlResolver.resolveUrl(did);
+
+        assertThat(resultApiUrl).isEmpty();
+
+    }
+
+    @Test
+    void resolveUrl_failureToGetDid() {
+
+        when(didResolver.resolve(did)).thenReturn(Result.failure("Failure"));
+
+        assertThatThrownBy(() -> urlResolver.resolveUrl(did)).isInstanceOf(CliException.class);
+    }
+
+    private DidDocument didDocument(String apiUrl) {
```

Comment:
nit: maybe we can take List<Services> services as argument here and this method can be reused to construct DidDocument without any service urls or with multiple service urls to test filtering as well. 

Just a suggestion but if you prefer some other way that's also fine. Thanks

Reply:
good, idea, changing